srml/im-online/src/lib.rs

```diff
@@ -67,44 +67,60 @@
 // Ensure we're `no_std` when compiling for Wasm.
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use app_crypto::RuntimeAppPublic;
+use app_crypto::{AppPublic, RuntimeAppPublic};
 use codec::{Encode, Decode};
 use primitives::offchain::{OpaqueNetworkState, StorageKind};
 use rstd::prelude::*;
 use session::historical::IdentificationTuple;
 use sr_io::Printable;
 use sr_primitives::{
 	Perbill, ApplyError,
-	traits::{Extrinsic as ExtrinsicT, Convert},
+	traits::{Convert, Extrinsic as ExtrinsicT, Member},
 	transaction_validity::{TransactionValidity, TransactionLongevity, ValidTransaction},
 };
 use sr_staking_primitives::{
 	SessionIndex, CurrentElectedSet,
 	offence::{ReportOffence, Offence, Kind},
 };
 use srml_support::{
-	StorageValue, decl_module, decl_event, decl_storage, StorageDoubleMap, print, ensure
+	decl_module, decl_event, decl_storage, print, ensure,
+	Parameter, StorageValue, StorageDoubleMap,
 };
 use system::ensure_none;
 
-mod app {
-	pub use app_crypto::sr25519 as crypto;
+mod app_sr25519 {
 	use app_crypto::{app_crypto, key_types::IM_ONLINE, sr25519};
-
 	app_crypto!(sr25519, IM_ONLINE);
 }
 
-/// A Babe authority keypair. Necessarily equivalent to the schnorrkel public key used in
-/// the main Babe module. If that ever changes, then this must, too.
-#[cfg(feature = "std")]
-pub type AuthorityPair = app::Pair;
+pub mod sr25519 {
+	/// An i'm online keypair using sr25519 as its crypto.
+	#[cfg(feature = "std")]
+	pub type AuthorityPair = super::app_sr25519::Pair;
+
+	/// An i'm online signature using sr25519 as its crypto.
+	pub type AuthoritySignature = super::app_sr25519::Signature;
+
+	/// An i'm online identifier using sr25519 as its crypto.
+	pub type AuthorityId = super::app_sr25519::Public;
+}
+
+mod app_ed25519 {
```

Comment:
much apologies for the mundane question just wondering why this module requires two different pairs?

Also, is this how we want to standardize it internally? a `mod app_xxx` and a `pub mod xxx`?

Reply:
It does not require two different pairs. It can just support both. There is no reason to just support `sr25519`.